mlrun/projects/pipelines.py

```diff
@@ -758,13 +763,20 @@ def run(
         namespace: str = None,
         source: str = None,
         notifications: list[mlrun.model.Notification] = None,
+        send_start_notification: bool = True,
     ) -> typing.Optional[_PipelineRunStatus]:
         workflow_name = normalize_workflow_name(name=name, project_name=project.name)
         workflow_id = None
 
         # for start message, fallback to old notification behavior
-        for notification in notifications or []:
-            project.notifiers.add_notification(notification.kind, notification.params)
+        # if a notification with `when=running` is provided, it will be used explicitly and others will be ignored
+        if send_start_notification:
+            for notification in notifications or []:
+                project.notifiers.add_notification(
+                    notification.kind, notification.params
+                )
+                if "running" in notification.when:
+                    break
```

Comment:
nitpicking, but maybe it is better to move the comment below
```suggestion
        if send_start_notification:
            for notification in notifications or []:
                project.notifiers.add_notification(
                    notification.kind, notification.params
                )
                 # if a notification with `when=running` is provided, it will be used explicitly and others will be ignored
                if "running" in notification.when:
                    break
```